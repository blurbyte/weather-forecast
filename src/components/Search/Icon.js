import styled from 'styled-components';

const Wrapper = styled.div`
  position: absolute;
  top: 0.8rem;
  right: 1rem;
  transition: color 0.35s linear;
`;

export default Wrapper;
