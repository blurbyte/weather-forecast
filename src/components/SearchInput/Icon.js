import styled from 'styled-components';

const Icon = styled.div`
  position: absolute;
  z-index: 9;
  top: 0.8rem;
  right: 1rem;
  transition: color 0.35s linear;
`;

export default Icon;
