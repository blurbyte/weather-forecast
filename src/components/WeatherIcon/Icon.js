// Renders appropriate weather conditions icon based on provided weather id

import React from 'react';
import PropTypes from 'prop-types';

import {
  BrokenClouds,
  ClearSkyDay,
  FewClouds,
  Mist,
  Rain,
  ScatteredClouds,
  ShowerRain,
  Snow,
  Thunderstorm
} from '../Icons';

const Icon = ({ weatherId }) => {
  if (weatherId >= 200 && weatherId < 300) {
    return <Thunderstorm />;
  } else if (weatherId >= 300 && weatherId < 400) {
    return <ShowerRain />;
  } else if (weatherId >= 500 && weatherId < 600) {
    return <Rain />;
  } else if (weatherId >= 600 && weatherId < 700) {
    return <Snow />;
  } else if (weatherId >= 700 && weatherId < 800) {
    return <Mist />;
  } else if (weatherId === 801) {
    return <FewClouds />;
  } else if (weatherId === 802) {
    return <ScatteredClouds />;
  } else if (weatherId === 803 || weatherId === 804) {
    return <BrokenClouds />;
  } else {
    return <ClearSkyDay />;
  }
};

Icon.propTypes = {
  weatherId: PropTypes.number.isRequired
};

export default Icon;
