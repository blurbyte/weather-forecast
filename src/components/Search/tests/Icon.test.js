import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Icon from '../Icon';

test('renders correctly', () => {
  const component = renderer.create(<Icon />);
  expect(component.toJSON()).toMatchSnapshot();
});
