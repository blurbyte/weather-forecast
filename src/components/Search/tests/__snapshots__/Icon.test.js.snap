// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  position: absolute;
  top: 0.8rem;
  right: 1rem;
  -webkit-transition: color 0.35s linear;
  transition: color 0.35s linear;
}

<div
  className="c0"
/>
`;
